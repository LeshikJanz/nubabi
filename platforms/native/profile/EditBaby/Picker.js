// @flow
import React, { Component } from 'react';
import {
  LayoutAnimation,
  Picker as PickerComponent,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
} from 'react-native';
import { range } from 'ramda';
import theme from '../../../../core/themes/defaultTheme';
import { isEditable } from '../../shared/forms';

export const weekOptions = range(20, 44).map(val => {
  return (
    <PickerComponent.Item
      label={val.toString()}
      value={val}
      key={val.toString()}
    />
  );
});

class Picker extends Component {
  props: {
    field: Object,
    onPickerOpen: (event: any) => void,
  };

  state = {
    displayed: false,
  };

  componentWillUpdate() {
    LayoutAnimation.easeInEaseOut();
  }

  toggle = (event: any) => {
    this.setState({ displayed: !this.state.displayed }, () => {
      this.props.onPickerOpen(event);
    });
  };

  renderPicker() {
    const { field } = this.props;

    return (
      <PickerComponent
        mode="dropdown"
        selectedValue={field.input.value}
        onValueChange={field.input.onChange}
        style={styles.pickerInput}
      >
        {weekOptions}
      </PickerComponent>
    );
  }

  render() {
    const { field } = this.props;
    const hasError = field.meta.touched && !!field.meta.error;
    const editable = isEditable(field);

    const Container = editable ? TouchableOpacity : View;

    const containerProps = editable ? { onPress: this.toggle } : {};

    const borderStyle = hasError ? { borderColor: theme.colors.danger } : {};

    return (
      <View style={{ flex: 1 }}>
        <Container
          {...containerProps}
          style={[styles.inputContainer, borderStyle]}
        >
          <Text>{this.props.field.input.value}</Text>
        </Container>
        {this.state.displayed ? this.renderPicker() : null}
      </View>
    );
  }
}

const styles = StyleSheet.create({
  inputContainer: {
    marginHorizontal: 30,
    // marginBottom: 15,
    borderBottomWidth: 1,
    paddingBottom: 5,
    borderColor: '#eff1f7',
  },
  pickerInput: {
    flex: 1,
    alignItems: 'stretch',
    backgroundColor: theme.colors.open.gray0,
    margin: 0,
    padding: 0,
  },
});

export default Picker;
