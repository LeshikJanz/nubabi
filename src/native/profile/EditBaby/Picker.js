// @flow
import React, { Component } from 'react';
import {
  View,
  Picker as PickerComponent,
  Text,
  TouchableOpacity,
  StyleSheet,
  LayoutAnimation,
} from 'react-native';
import { range } from 'ramda';
import theme from '../../../common/themes/defaultTheme';

class Picker extends Component {
  props: {
    field: Object,
    onPickerOpen: (event: any) => void,
  };

  state = {
    displayed: false,
  };

  componentWillUpdate() {
    LayoutAnimation.easeInEaseOut();
  }

  toggle = (event: any) => {
    this.setState({ displayed: !this.state.displayed }, () => {
      this.props.onPickerOpen(event);
    });
  };

  renderPicker() {
    const { field } = this.props;

    const weekOptions = range(32, 43).map(val => {
      return (
        <PickerComponent.Item
          label={val.toString()}
          value={val}
          key={val.toString()}
        />
      );
    });

    return (
      <PickerComponent
        mode="dropdown"
        selectedValue={field.input.value}
        onValueChange={field.input.onChange}
        style={styles.pickerInput}
      >
        {weekOptions}
      </PickerComponent>
    );
  }

  render() {
    return (
      <View style={{ flex: 1 }}>
        <TouchableOpacity onPress={this.toggle} style={styles.inputContainer}>
          <Text>{this.props.field.input.value}</Text>
        </TouchableOpacity>
        {this.state.displayed ? this.renderPicker() : null}
      </View>
    );
  }
}

const styles = StyleSheet.create({
  inputContainer: {
    marginHorizontal: 30,
    // marginBottom: 15,
    borderBottomWidth: 1,
    paddingBottom: 5,
    borderColor: '#eff1f7',
  },
  pickerInput: {
    flex: 1,
    alignItems: 'stretch',
    backgroundColor: theme.colors.open.gray0,
    margin: 0,
    padding: 0,
  },
});

export default Picker;
