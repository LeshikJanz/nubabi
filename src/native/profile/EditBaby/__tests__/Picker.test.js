import 'react-native';
import React from 'react';
import Picker from '../Picker';
import renderer from 'react-test-renderer';

test('it renders correctly', () => {
  const mockField = {
    input: {
      value: 30,
    },
    meta: {
      error: false,
    },
  };
  const tree = renderer.create(<Picker field={mockField} />).toJSON();

  expect(tree).toMatchSnapshot();
});
