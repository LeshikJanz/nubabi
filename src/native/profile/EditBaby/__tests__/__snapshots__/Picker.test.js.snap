// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
style = Object {}

<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#eff1f7",
        "marginHorizontal": 30,
        "opacity": 1,
        "paddingBottom": 5,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
    >
      30
    </Text>
  </View>
</View>
`;
